### PR TITLE
Dannyfs feature

### PR DESCRIPTION
**Pull request by DannyFGitHub**
Changes to BorrowBookUI, BorrowBookControl

### CODE REVIEW

File: BorrowBookControl.java [hunk 1]
```diff
@@ -2,116 +2,118 @@
...
+        if (!state.equals(CONTROL_STATE.INITIALISED)) {
+            throw new RuntimeException("BorrowBookControl: cannot call setUI except in INITIALISED state");
+        }
+        this.ui = ui;
+        ui.setState(BorrowBookUI.UI_STATE.READY);
+        state = CONTROL_STATE.READY;
+    }
+
+
+    public void cardSwiped(int memberId) {
```
Comment (ablewitt) [inline BorrowBookControl.java]: Issue No.2
Acronym in memberId should memberID

File: BorrowBookControl.java [hunk 2]
```diff
@@ -2,116 +2,118 @@
...
+            state = CONTROL_STATE.SCANNING;
+        }
+        else {
+            ui.display("Member cannot borrow at this time");
+            ui.setState(BorrowBookUI.UI_STATE.RESTRICTED);
+        }
+    }
+
+
+    public void bookScanned(int bookId) {
```
Comment (ablewitt) [inline BorrowBookControl.java]: Issue No.3
Upper case acronym needed in bookId

File: BorrowBookUI.java [hunk 3]
```diff
@@ -2,116 +2,116 @@
...
+
+            switch (state) {
+
+                case CANCELLED:
+                    printOutput("Borrowing Cancelled");
+                    return;
+
+
+                case READY:
+                    String memberIdStr = readInput("Swipe member card (press <enter> to cancel): ");
```
Comment (ablewitt) [inline BorrowBookUI.java]: Issue No.5
Uppercase acronym needed in memberIdStr, you could also drop the Str for Input as the type is pretty obvious

File: BorrowBookControl.java [hunk 4]
```diff
@@ -7,8 +7,8 @@
 
     private library library;
     private member member;
-    private enum CONTROL_STATE { INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED };
-    private CONTROL_STATE state;
+    private enum controlState { INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED };
```
Comment (ablewitt) [inline BorrowBookControl.java]: Issue no. 1
enum should be ControlState style rule #3

File: BorrowBookUI.java [hunk 5]
```diff
@@ -3,17 +3,17 @@
 
 public class BorrowBookUI {
 
-    public static enum UI_STATE { INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED };
+    public static enum uiState { INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED };
```
Comment (ablewitt) [inline BorrowBookUI.java]: Issue no. 1
enum should be UIState style rule #3 and rule #12

**Comment by danielnetwork** [comment]
Merged dannyfs_feature with master locally without any conflicts. Successfully run the app, created several new users and books. Was able to take books out for loan and return them.

**Pull request merged**